ogre2/src/Ogre2Scene.cc

```diff
@@ -567,3 +569,39 @@ bool Ogre2Scene::ShadowsDirty() const
 {
   return this->dataPtr->shadowsDirty;
 }
+
+//////////////////////////////////////////////////
+void Ogre2Scene::SetSkyEnabled(bool _enabled)
+{
+  MaterialPtr skyboxMat;
+  if (_enabled)
+  {
+    // get skybox material
+    std::string skyboxMatName = "Default/skybox";
+    skyboxMat = this->Material(skyboxMatName);
+    if (!skyboxMat)
+    {
+      // ogre2 should be able to find this texture as resource search
+      // paths are already set up in Ogre2RenderEngine.cc
+      std::string skyboxEnvMap = "skybox.dds";
+      skyboxMat = this->CreateMaterial(skyboxMatName);
+      skyboxMat->SetEnvironmentMap(skyboxEnvMap);
+    }
+  }
+  this->SetBackgroundMaterial(skyboxMat);
+  for (unsigned int i = 0; i < this->Sensors()->Size(); ++i)
+  {
+    auto sensor = this->Sensors()->GetByIndex(i);
+    auto camera = std::dynamic_pointer_cast<Ogre2Camera>(sensor);
+    if (camera)
+    {
+      camera->SetBackgroundMaterial(skyboxMat);
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+bool Ogre2Scene::SkyEnabled() const
+{
+  return this->BackgroundMaterial() != nullptr;
```

Comment:
Is it possible that the user sets a different background material using `SetBackgroundMaterial`, and this returns true even though the sky is not enabled?

Reply:
added a variable to keep track of sky state 0823885